### ISSUE TITLE
Cloaking on 3.5.0 is sometimes not applied

### ISSUE THREAD

**Issue by DarkTexas** [inspircd/inspircd#1773]
**Description**

Some Users do not seem to receive a cloak. This behaviour appears to be random. It happens that one user joins cloaked for one but another user sees the user with their real ip.

**Steps to reproduce the issue:**

- no reproduction steps available -

**Describe the results you received:**

Some users are seing real ips of other users.

**Describe the results you expected:**

All users see a cloaked IP.

**Additional information you deem important (e.g. issue happens only occasionally):**

The issue appears randomly and seems to be bound to many factors.

InspIRCd-3.5.0.

Modules:

```
[09.Apr.20 18:35:00] core_channel.so InspIRCd-3.5.0 V-- Provides the INVITE, JOIN, KICK, NAMES, and TOPIC commands
[09.Apr.20 18:35:00] core_dns.so InspIRCd-3.5.0 V-- Provides support for DNS lookups
[09.Apr.20 18:35:00] core_hostname_lookup.so InspIRCd-3.5.0 V-- Provides support for DNS lookups on connecting clients
[09.Apr.20 18:35:00] core_info.so InspIRCd-3.5.0 V-- Provides the ADMIN, COMMANDS, INFO, MODULES, MOTD, TIME, and VERSION commands
[09.Apr.20 18:35:00] core_list.so InspIRCd-3.5.0 V-- Provides the LIST command
[09.Apr.20 18:35:00] core_loadmodule.so InspIRCd-3.5.0 V-- Provides the LOADMODULE and UNLOADMODULE commands
[09.Apr.20 18:35:00] core_lusers.so InspIRCd-3.5.0 V-- Provides the LUSERS command
[09.Apr.20 18:35:00] core_message.so InspIRCd-3.5.0 V-- Provides the NOTICE, PRIVMSG, and SQUERY commands
[09.Apr.20 18:35:00] core_mode.so InspIRCd-3.5.0 V-- Provides the MODE command
[09.Apr.20 18:35:00] core_oper.so InspIRCd-3.5.0 V-- Provides the DIE, KILL, OPER, REHASH, and RESTART commands
[09.Apr.20 18:35:00] core_reloadmodule.so InspIRCd-3.5.0 V-- Provides the RELOADMODULE command
[09.Apr.20 18:35:00] core_serialize_rfc.so InspIRCd-3.5.0 V-- RFC client protocol serializer and unserializer
[09.Apr.20 18:35:00] core_stats.so InspIRCd-3.5.0 V-- Provides the STATS command
[09.Apr.20 18:35:00] core_stub.so InspIRCd-3.5.0 V-- Provides stubs for unimplemented commands
[09.Apr.20 18:35:00] core_user.so InspIRCd-3.5.0 V-- Provides the AWAY, ISON, NICK, PART, PASS, PING, PONG, QUIT, USERHOST, and USER commands
[09.Apr.20 18:35:00] core_wallops.so InspIRCd-3.5.0 V-- Provides the WALLOPS command
[09.Apr.20 18:35:00] core_who.so InspIRCd-3.5.0 V-- Provides the WHO command
[09.Apr.20 18:35:00] core_whois.so InspIRCd-3.5.0 V-- Provides the WHOIS command
[09.Apr.20 18:35:00] core_whowas.so InspIRCd-3.5.0 V-- Provides the WHOWAS command
[09.Apr.20 18:35:00] core_xline.so InspIRCd-3.5.0 V-- Provides the ELINE, GLINE, KLINE, QLINE, and ZLINE commands
[09.Apr.20 18:35:00] m_alias.so InspIRCd-3.5.0 V-- Provides aliases of commands
[09.Apr.20 18:35:00] m_allowinvite.so InspIRCd-3.5.0 V-- Provides channel mode +A to allow /INVITE freely on a channel, and extban 'A' to deny specific users it
[09.Apr.20 18:35:00] m_alltime.so InspIRCd-3.5.0 V-O Provides the ALLTIME command, displays timestamps from all servers connected to the network
[09.Apr.20 18:35:00] m_autoop.so InspIRCd-3.5.0 V-- Provides channel mode +w, basic channel access controls
[09.Apr.20 18:35:00] m_banexception.so InspIRCd-3.5.0 V-- Provides channel mode +e, ban exceptions
[09.Apr.20 18:35:00] m_banredirect.so InspIRCd-3.5.0 VC- Allows an extended ban (+b) syntax redirecting banned users to another channel
[09.Apr.20 18:35:00] m_bcrypt.so InspIRCd-3.5.0 V-- Implements bcrypt hashing
[09.Apr.20 18:35:00] m_blockcaps.so InspIRCd-3.5.0 V-- Provides support to block all-CAPS channel messages and notices
[09.Apr.20 18:35:00] m_blockcolor.so InspIRCd-3.5.0 V-- Provides channel mode +c to block color
[09.Apr.20 18:35:00] m_botmode.so InspIRCd-3.5.0 V-- Provides user mode +B to mark the user as a bot
[09.Apr.20 18:35:00] m_callerid.so InspIRCd-3.5.0 VC- Implementation of callerid, provides user mode +g and the ACCEPT command
[09.Apr.20 18:35:00] m_cap.so InspIRCd-3.5.0 V-- Provides support for CAP capability negotiation
[09.Apr.20 18:35:00] m_cban.so InspIRCd-3.5.0 VC- Provides the CBAN command, like Q-lines, but for channels
[09.Apr.20 18:35:00] m_censor.so InspIRCd-3.5.0 V-- Provides user and channel mode +G
[09.Apr.20 18:35:00] m_cgiirc.so InspIRCd-3.5.0 V-- Enables forwarding the real IP address of a user from a gateway to the IRC server
[09.Apr.20 18:35:00] m_chanfilter.so InspIRCd-3.5.0 V-- Provides channel-specific censor lists (like mode +G but varies from channel to channel)
[09.Apr.20 18:35:00] m_chanhistory.so InspIRCd-3.5.0 V-- Provides channel mode +H, allows for the channel message history to be replayed on join
[09.Apr.20 18:35:00] m_chanlog.so InspIRCd-3.5.0 V-- Logs snomask output to channel(s)
[09.Apr.20 18:35:00] m_channelban.so InspIRCd-3.5.0 V-O Provides extban 'j', ban users that are present in another channel, and optionally on their status there
[09.Apr.20 18:35:00] m_check.so InspIRCd-3.5.0 V-O Provides the CHECK command to view user, channel, IP address or hostname information
[09.Apr.20 18:35:00] m_checkbans.so InspIRCd-3.5.0 --- Gives /checkbans, /testban, and /whyban - channel ban helper commands.
[09.Apr.20 18:35:00] m_chghost.so InspIRCd-3.5.0 V-O Provides the CHGHOST command
[09.Apr.20 18:35:00] m_chgident.so InspIRCd-3.5.0 V-O Provides the CHGIDENT command
[09.Apr.20 18:35:00] m_chgname.so InspIRCd-3.5.0 V-O Provides the CHGNAME command
[09.Apr.20 18:35:00] m_classban.so InspIRCd-3.5.0 V-O Provides extban 'n', connection class bans
[09.Apr.20 18:35:00] m_clearchan.so InspIRCd-3.5.0 V-O Provides the CLEARCHAN command that allows opers to masskick, masskill or mass G/Z-line users on a channel
[09.Apr.20 18:35:00] m_cloaking.so InspIRCd-3.5.0 VC- Provides masking of user hostnames
[09.Apr.20 18:35:00] m_clones.so InspIRCd-3.5.0 V-- Provides the CLONES command to retrieve information on clones
[09.Apr.20 18:35:00] m_commonchans.so InspIRCd-3.5.0 V-- Provides user mode +c, requires users to share a common channel with you to private message you
[09.Apr.20 18:35:00] m_conn_join.so InspIRCd-3.5.0 V-- Forces users to join the specified channel(s) on connect
[09.Apr.20 18:35:00] m_conn_umodes.so InspIRCd-3.5.0 V-- Sets (and unsets) modes on users when they connect
[09.Apr.20 18:35:00] m_conn_vhost.so InspIRCd-3.5.0 --- Sets a connect block configured vhost on users when they connect
[09.Apr.20 18:35:00] m_conn_waitpong.so InspIRCd-3.5.0 V-- Require pong prior to registration
[09.Apr.20 18:35:00] m_connectban.so InspIRCd-3.5.0 V-- Throttles the connections of IP ranges who try to connect flood
[09.Apr.20 18:35:00] m_connflood.so InspIRCd-3.5.0 V-- Connection throttle
[09.Apr.20 18:35:00] m_delayjoin.so InspIRCd-3.5.0 V-- Provides channel mode +D, delay-join, users don't appear as joined to others until they speak
[09.Apr.20 18:35:00] m_delaymsg.so InspIRCd-3.5.0 V-- Provides channel mode +d <int>, to deny messages to a channel until <int> seconds have passed
[09.Apr.20 18:35:00] m_denychans.so InspIRCd-3.5.0 V-- Implements config tags which allow blocking of joins to channels
[09.Apr.20 18:35:00] m_disable.so InspIRCd-3.5.0 V-- Provides support for disabling commands and modes
[09.Apr.20 18:35:00] m_dnsbl.so InspIRCd-3.5.0 V-- Provides handling of DNS blacklists
[09.Apr.20 18:35:00] m_exemptchanops.so InspIRCd-3.5.0 V-- Provides the ability to allow channel operators to be exempt from certain modes
[09.Apr.20 18:35:00] m_filter.so InspIRCd-3.5.0 VC- Provides text (spam) filtering
[09.Apr.20 18:35:00] m_flashpolicyd.so InspIRCd-3.5.0 V-- Flash Policy Daemon, allows Flash IRC clients to connect
[09.Apr.20 18:35:00] m_gecosban.so InspIRCd-3.5.0 V-O Provides a way to ban users by their real name with the 'a' and 'r' extbans
[09.Apr.20 18:35:00] m_geocmd.so InspIRCd-3.5.0 --- Provides the /GEOLOCATE command which performs Geolocation lookups on arbitrary IP addresses
[09.Apr.20 18:35:00] m_globalload.so InspIRCd-3.5.0 VC- Allows global loading of a module
[09.Apr.20 18:35:00] m_globops.so InspIRCd-3.5.0 V-- Provides the GLOBOPS command and snomask 'g'
[09.Apr.20 18:35:00] m_helpop.so InspIRCd-3.5.0 V-- Provides help to users via the HELPOP command
[09.Apr.20 18:35:00] m_hidechans.so InspIRCd-3.5.0 V-- Provides support for hiding channels with user mode +I
[09.Apr.20 18:35:00] m_hideidle.so InspIRCd-3.5.0 --- Provides the +a usermode that hides idle and signon time in WHOIS from non-opers
[09.Apr.20 18:35:00] m_hideoper.so InspIRCd-3.5.0 V-- Provides support for hiding oper status with user mode +H
[09.Apr.20 18:35:00] m_hostchange.so InspIRCd-3.5.0 V-- Provides rule-based masking of user hostnames
[09.Apr.20 18:35:00] m_hostcycle.so InspIRCd-3.5.0 V-- Cycles users in all their channels when their host or ident changes
[09.Apr.20 18:35:00] m_inviteexception.so InspIRCd-3.5.0 V-- Provides channel mode +I, invite exceptions
[09.Apr.20 18:35:00] m_ircv3.so InspIRCd-3.5.0 V-- Provides support for extended-join, away-notify and account-notify CAP capabilities
[09.Apr.20 18:35:00] m_ircv3_accounttag.so InspIRCd-3.5.0 V-- Provides the account-tag IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_batch.so InspIRCd-3.5.0 V-- Provides the batch IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_capnotify.so InspIRCd-3.5.0 V-- Provides the cap-notify IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_chghost.so InspIRCd-3.5.0 V-- Provides the chghost IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_ctctags.so InspIRCd-3.5.0 VC- Provides the message-tags IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_echomessage.so InspIRCd-3.5.0 V-- Provides the echo-message IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_invitenotify.so InspIRCd-3.5.0 V-- Provides the invite-notify IRCv3 extension
[09.Apr.20 18:35:00] m_ircv3_msgid.so InspIRCd-3.5.0 V-- Provides the msgid IRCv3 tag
[09.Apr.20 18:35:00] m_ircv3_servertime.so InspIRCd-3.5.0 V-- Provides the server-time IRCv3 extension
[09.Apr.20 18:35:00] m_joinflood.so InspIRCd-3.5.0 V-- Provides channel mode +j, join flood protection
[09.Apr.20 18:35:00] m_joinpartsno.so InspIRCd-3.5.0 --- Creates SNOMask for user joins/parts
[09.Apr.20 18:35:00] m_joinpartspam.so InspIRCd-3.5.0 --O Provides channel mode +x for blocking Join/Part spammers.
[09.Apr.20 18:35:00] m_jumpserver.so InspIRCd-3.5.0 --- Provides support for the RPL_REDIR numeric and the /JUMPSERVER command.
[09.Apr.20 18:35:00] m_kicknorejoin.so InspIRCd-3.5.0 VC- Provides channel mode +J, delays rejoins after kicks
[09.Apr.20 18:35:00] m_knock.so InspIRCd-3.5.0 V-O Provides the KNOCK command and channel mode +K
[09.Apr.20 18:35:00] m_lockserv.so InspIRCd-3.5.0 V-- Provides the LOCKSERV and UNLOCKSERV commands to lock the server and block all incoming connections until unlocked again
[09.Apr.20 18:35:00] m_maphide.so InspIRCd-3.5.0 V-- Replaces the output of the MAP and LINKS commands with an URL
[09.Apr.20 18:35:00] m_md5.so InspIRCd-3.5.0 V-- Implements MD5 hashing
[09.Apr.20 18:35:00] m_messageflood.so InspIRCd-3.5.0 V-- Provides channel mode +f, message flood protection
[09.Apr.20 18:35:00] m_modenotice.so InspIRCd-3.5.0 V-- Provides the MODENOTICE command
[09.Apr.20 18:35:00] m_monitor.so InspIRCd-3.5.0 V-- Provides MONITOR support
[09.Apr.20 18:35:00] m_muteban.so InspIRCd-3.5.0 V-O Provides extban 'm', mute bans
[09.Apr.20 18:35:00] m_mysql.so InspIRCd-3.5.0 V-- Provides MySQL support
[09.Apr.20 18:35:00] m_namedstats.so InspIRCd-3.5.0 --- Allows /STATS queries by name
[09.Apr.20 18:35:00] m_namesx.so InspIRCd-3.5.0 V-- Provides the NAMESX (CAP multi-prefix) capability
[09.Apr.20 18:35:00] m_nickdelay.so InspIRCd-3.5.0 --- Enforces a delay between nick changes per user
[09.Apr.20 18:35:00] m_nicklock.so InspIRCd-3.5.0 V-O Provides the NICKLOCK command, allows an oper to change a users nick and lock them to it until they quit
[09.Apr.20 18:35:00] m_noctcp.so InspIRCd-3.5.0 V-- Provides user mode +T and channel mode +C to block CTCPs
[09.Apr.20 18:35:00] m_nokicks.so InspIRCd-3.5.0 V-- Provides channel mode +Q to prevent kicks on the channel
[09.Apr.20 18:35:00] m_nonicks.so InspIRCd-3.5.0 V-- Provides channel mode +N and extban 'N' which prevents nick changes on the channel
[09.Apr.20 18:35:00] m_nonotice.so InspIRCd-3.5.0 V-- Provides channel mode +T to block notices to the channel
[09.Apr.20 18:35:00] m_nopartmsg.so InspIRCd-3.5.0 V-O Provides extban 'p', part message bans
[09.Apr.20 18:35:00] m_noprivatemode.so InspIRCd-3.5.0 --O Sets channel mode +s (secret) when users try to set channel mode +p (private).
[09.Apr.20 18:35:00] m_nouidnick.so InspIRCd-3.5.0 --- Disallows changing nick to UID using /NICK
[09.Apr.20 18:35:00] m_operchans.so InspIRCd-3.5.0 V-- Provides support for oper-only channels via channel mode +O and extban 'O'
[09.Apr.20 18:35:00] m_operjoin.so InspIRCd-3.5.0 V-- Forces opers to join the specified channel(s) on oper-up
[09.Apr.20 18:35:00] m_operlog.so InspIRCd-3.5.0 V-- Provides logging of all oper commands to the ircd log at the default loglevel
[09.Apr.20 18:35:00] m_opermodes.so InspIRCd-3.5.0 V-- Sets (and unsets) modes on opers when they oper up
[09.Apr.20 18:35:00] m_operprefix.so InspIRCd-3.5.0 V-- Gives opers channel mode +y which provides a staff prefix
[09.Apr.20 18:35:00] m_override.so InspIRCd-3.5.0 V-- Provides support for allowing opers to override certain things
[09.Apr.20 18:35:00] m_password_hash.so InspIRCd-3.5.0 V-- Provides the ability to hash passwords to other modules
[09.Apr.20 18:35:00] m_permchannels.so InspIRCd-3.5.0 V-- Provides channel mode +P to provide permanent channels
[09.Apr.20 18:35:00] m_redirect.so InspIRCd-3.5.0 V-- Provides channel mode +L (limit redirection) and user mode +L (no forced redirection)
[09.Apr.20 18:35:00] m_regex_glob.so InspIRCd-3.5.0 V-- Regex provider module using plain wildcard matching
[09.Apr.20 18:35:00] m_regex_pcre.so InspIRCd-3.5.0 V-- Regex Provider Module for PCRE
[09.Apr.20 18:35:00] m_remove.so InspIRCd-3.5.0 V-O Provides the REMOVE command as an alternative to KICK, it makes users appear to have left the channel
[09.Apr.20 18:35:00] m_rmode.so InspIRCd-3.5.0 V-- Allows glob-based removal of list modes
[09.Apr.20 18:35:00] m_sajoin.so InspIRCd-3.5.0 V-O Provides the SAJOIN command, allows opers to force-join users to channels
[09.Apr.20 18:35:00] m_sakick.so InspIRCd-3.5.0 V-O Provides the SAKICK command
[09.Apr.20 18:35:00] m_samode.so InspIRCd-3.5.0 V-- Provides the SAMODE command, allows opers to change modes on channels and users
[09.Apr.20 18:35:00] m_sanick.so InspIRCd-3.5.0 V-O Provides the SANICK command, allows opers to change the nicknames of users
[09.Apr.20 18:35:00] m_sapart.so InspIRCd-3.5.0 V-O Provides the SAPART command, allows opers to force-part users from channels
[09.Apr.20 18:35:00] m_saquit.so InspIRCd-3.5.0 V-O Provides the SAQUIT command, allows opers to force-quit users
[09.Apr.20 18:35:00] m_sasl.so InspIRCd-3.5.0 V-- Provides support for IRC Authentication Layer (aka: SASL) via AUTHENTICATE
[09.Apr.20 18:35:00] m_satopic.so InspIRCd-3.5.0 V-- Provides the SATOPIC command
[09.Apr.20 18:35:00] m_seenicks.so InspIRCd-3.5.0 V-- Provides snomasks 'n' and 'N' to see local and remote nickchanges
[09.Apr.20 18:35:00] m_serverban.so InspIRCd-3.5.0 V-O Provides extban 's' to ban users connected to a specified server
[09.Apr.20 18:35:00] m_services_account.so InspIRCd-3.5.0 V-O Provides support for ircu-style services accounts, including channel mode +R, etc
[09.Apr.20 18:35:00] m_servprotect.so InspIRCd-3.5.0 V-- Provides user mode +k to protect services from kicks, kills, and mode changes
[09.Apr.20 18:35:00] m_sethost.so InspIRCd-3.5.0 V-- Provides the SETHOST command
[09.Apr.20 18:35:00] m_setident.so InspIRCd-3.5.0 V-- Provides the SETIDENT command
[09.Apr.20 18:35:00] m_setidle.so InspIRCd-3.5.0 V-- Provides the SETIDLE command, allows opers to set their idle time
[09.Apr.20 18:35:00] m_setname.so InspIRCd-3.5.0 V-- Provides the SETNAME command
[09.Apr.20 18:35:00] m_sha1.so InspIRCd-3.5.0 V-- Implements SHA-1 hashing
[09.Apr.20 18:35:00] m_sha256.so InspIRCd-3.5.0 V-- Implements SHA-256 hashing
[09.Apr.20 18:35:00] m_showwhois.so InspIRCd-3.5.0 V-O Provides user mode +W for opers to see when a user uses WHOIS on them
[09.Apr.20 18:35:00] m_shun.so InspIRCd-3.5.0 VC- Provides the SHUN command, which stops a user from executing all except configured commands
[09.Apr.20 18:35:00] m_silence.so InspIRCd-3.5.0 V-O Provides support for blocking users with the SILENCE command
[09.Apr.20 18:35:00] m_slowmode.so InspIRCd-3.5.0 -C- Provides channel mode +W (slowmode)
[09.Apr.20 18:35:00] m_spanningtree.so InspIRCd-3.5.0 V-- Allows servers to be linked
[09.Apr.20 18:35:00] m_sqlauth.so InspIRCd-3.5.0 V-- Allow/deny connections based upon an arbitrary SQL table
[09.Apr.20 18:35:00] m_ssl_openssl.so InspIRCd-3.5.0 V-- Provides SSL support via OpenSSL
[09.Apr.20 18:35:00] m_sslinfo.so InspIRCd-3.5.0 V-- SSL Certificate Utilities
[09.Apr.20 18:35:00] m_sslrehashsignal.so InspIRCd-3.5.0 V-- Reloads SSL credentials on SIGUSR1
[09.Apr.20 18:35:00] m_stripcolor.so InspIRCd-3.5.0 V-- Provides channel mode +S, strip ansi color
[09.Apr.20 18:35:00] m_svshold.so InspIRCd-3.5.0 VC- Implements SVSHOLD, like Q-lines, but can only be added/removed by Services
[09.Apr.20 18:35:00] m_swhois.so InspIRCd-3.5.0 V-O Provides the SWHOIS command which allows setting of arbitrary WHOIS lines
[09.Apr.20 18:35:00] m_timedbans.so InspIRCd-3.5.0 VC- Provides the TBAN command, timed channel bans
[09.Apr.20 18:35:00] m_tline.so InspIRCd-3.5.0 V-- Provides the TLINE command, used to test how many users a mask matches against
[09.Apr.20 18:35:00] m_uhnames.so InspIRCd-3.5.0 V-- Provides the UHNAMES (CAP userhost-in-names) capability
[09.Apr.20 18:35:00] m_uninvite.so InspIRCd-3.5.0 V-O Provides the UNINVITE command which lets users un-invite other users from channels
[09.Apr.20 18:35:00] m_userip.so InspIRCd-3.5.0 V-- Provides the USERIP command
[09.Apr.20 18:35:00] m_vhost.so InspIRCd-3.5.0 V-- Provides masking of user hostnames via the VHOST command
[09.Apr.20 18:35:00] m_watch.so InspIRCd-3.5.0 V-- Provides WATCH support
[09.Apr.20 18:35:00] m_xline_db.so InspIRCd-3.5.0 V-- Provides the ability to store X-lines in a database file
[09.Apr.20 18:35:00] m_xlinetools.so InspIRCd-3.5.0 --- X-line management tools
[09.Apr.20 18:35:00] End of MODULES list
```

**Comment by genius3000** [comment]
Few questions
1) Do the users that don't see a cloak see an IP address or a hostname?
2) Is there any common factor to who sees a cloak and who doesn't (local vs remote server)?

With the users that appear to miss the cloak to some: 
1) Are they hitting a connect class that uses m_conn_vhost?
2) Do they still have the +x user mode?
3) Can you get client logs, perhaps even raw logs of when they connect? Be nice to see the server messages of +x, host being set, etc.

**Comment by DarkTexas** [comment]
1. A Hostmask, example:

User a saw) [17:44:34] * Michael (54876251@xremovedx.dip0.t-ipconnect.de) has joined #germany
User b saw) [17:44:34] * Michael (54876251@65h.6ep.dhgu71.IP) has joined #Germany

2. When I checked on one user I did not see the "real ip/hostname" on the remote servers, when I connected to the server the user was on I saw its real ip. But that's all I was able to find out yet.

---

1. All connect classes are forced to use +x and +x is also locked on the server, I have not checked the user modes of a affected user. Will do that next time it happens.

very unlikely :-/

**Comment by DarkTexas** [comment]
Some more information as it happened today. For testing reasons I have unloaded the m_conn_vhost module yesterday, so it's not that.

However, a User spotted one of those "uncloaked" users:

(22:20:55) * Joins: n00b (5cdad2fc@ipserXXXXXXXXXXXXXX218.pools.vodafone-ip.de) 

The User who was able to spot this join was on the same server as the user with the missing cloaking. From my side, on another server it was perfectly fine.

When I connected to the Server the User was on I did receive a cloak correctly and /whoised the user, I did see the User without any cloaking as well. Now, for testing reasons I opered up, and the User had the correct mode(s):

n00b is using modes +Tiwx

Its not a issue occuring on one Server only, it's random on different servers, but one behaviour always seems to be the same: this is a local issue, global the cloak appears to work.

**Comment by genius3000** [comment]
This looks like `core_hostname_resolution` is sometimes changing the real and display hosts after cloaking has been applied. The issue of `core_hostname_resolution` changing the display host when it shouldn't be has been fixed in https://github.com/inspircd/inspircd/commit/d6133886f5a84000fcb2df6e993d252ecb55b72d and likely will resolve this. If you're willing to test, the simplest way would be to disable `resolvehostnames` in the connect classes. Better than that would be to upgrade or at least apply that single commit and reload the module, see if that does fix it.

**Comment by DarkTexas** [comment]
> This looks like `core_hostname_resolution` is sometimes changing the real and display hosts after cloaking has been applied. The issue of `core_hostname_resolution` changing the display host when it shouldn't be has been fixed in [d613388](https://github.com/inspircd/inspircd/commit/d6133886f5a84000fcb2df6e993d252ecb55b72d) and likely will resolve this. If you're willing to test, the simplest way would be to disable `resolvehostnames` in the connect classes. Better than that would be to upgrade or at least apply that single commit and reload the module, see if that does fix it.

I do think this issue is resolved with changing what you said. Still looking for any users joining without one, haven't seen any so far I think.
